db/consistency_level.cc

```diff
@@ -334,7 +334,13 @@ filter_for_query(consistency_level cl,
         if (!old_node && ht_max - ht_min > 0.01) { // if there is old node or hit rates are close skip calculations
             // local node is always first if present (see storage_proxy::get_endpoints_for_reading)
             unsigned local_idx = erm.get_topology().is_me(epi[0].first) ? 0 : epi.size() + 1;
-            live_endpoints = boost::copy_range<inet_address_vector_replica_set>(miss_equalizing_combination(epi, local_idx, remaining_bf, bool(extra)));
+            auto weighted = boost::copy_range<inet_address_vector_replica_set>(miss_equalizing_combination(epi, local_idx, remaining_bf, bool(extra)));
+            // Workaround for https://github.com/scylladb/scylladb/issues/9285
+            auto last = std::adjacent_find(weighted.begin(), weighted.end());
```

Comment:
Is it guaranteed that duplicates will be adjacent?

Reply:
Maybe we should push the workaround into miss_equalizing_combination?

Reply:
As far as I saw where the problem is cumming from yes. @nyh?

Reply:
> Maybe we should push the workaround into miss_equalizing_combination?

Why? It does not have original `live_endpoints`. It will just make it more complicated.

Reply:
@avikivity yes, the way ssample() ("Systematic sampling") works is that if an item has probability >1/K, it can be included more than once in the output list but adjacent. You can see an explanation of systematic sampling in a comment in heat_load_balance.cc.

@gleb-cloudius what miss_equalizing_combination could have done is not to revert to the original live_endpints but just to drop the extra node and come up with a new random one, just like we did with _extra. I admit it's more code.